File: aspnet-core/src/HC.WeChat.Core/Retailers/ShopReportData.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace HC.WeChat.Retailers
{
    public class ShopReportData
    {
        /// <summary>
        /// RootId
        /// </summary>
        public int RootId { get; set; }

        /// <summary>
        /// CompanyId
        /// </summary>
        public int CompanyId { get; set; }

        /// <summary>
        /// AreaId
        /// </summary>
        public int AreaId { get; set; }

        /// <summary>
        /// SlsmanNameId
        /// </summary>
        public int SlsmanNameId { get; set; }

        /// <summary>
        /// GroupNum
        /// </summary>
        public int GroupNum { get; set; }

        /// <summary>
        /// Organization
        /// </summary>
        public string Organization { get; set; }

        /// <summary>
        /// ShopTotal
        /// </summary>
        public int? ShopTotal { get; set; }

        /// <summary>
        /// ScanQuantity
        /// </summary>
        public int? ScanQuantity { get; set; }

        /// <summary>
        /// ScanFrequency
        /// </summary>
        public int? ScanFrequency { get; set; }

        /// <summary>
        /// PriceTotal
        /// </summary>
        public decimal? PriceTotal { get; set; }

        /// <summary>
        /// CustIntegral
        /// </summary>
        public int? CustIntegral { get; set; }

        /// <summary>
        /// RetailerIntegral
        /// </summary>
        public int? RetailerIntegral { get; set; }

        /// <summary>
        /// CreationTime
        /// </summary>
        public DateTime? CreationTime { get; set; }

        public string Specification { get; set; }
    }
}

File: ng-admin/src/app/shared/service-proxies/entity/shopReportData.ts
export class ShopReportData implements IShopReportData {
    rootId: number;
    companyId: number;
    areaId: number;
    slsmanNameId: number;
    groupNum: number;
    organization: string;
    shopTotal: number;
    scanQuantity: number;
    scanFrequency: number;
    priceTotal: number;
    custIntegral: number;
    retailerIntegral: number;
    creationTime: Date;
    specification: string;
    constructor(data?: IShopReportData) {
        if (data) {
            for (var property in data) {
                if (data.hasOwnProperty(property))
                    (<any>this)[property] = (<any>data)[property];
            }
        }
    }

    init(data?: any) {
        if (data) {
            this.rootId = data["rootId"];
            this.companyId = data["companyId"];
            this.areaId = data["areaId"];
            this.slsmanNameId = data["slsmanNameId"];
            this.groupNum = data["groupNum"];
            this.organization = data["organization"];
            this.shopTotal = data["shopTotal"];
            this.scanQuantity = data["scanQuantity"];
            this.scanFrequency = data["scanFrequency"];
            this.priceTotal = data["priceTotal"];
            this.custIntegral = data["custIntegral"];
            this.retailerIntegral = data["retailerIntegral"];
            this.creationTime = data["creationTime"];
            this.specification = data["specification"];
        }
    }

    static fromJS(data: any): ShopReportData {
        let result = new ShopReportData();
        result.init(data);
        return result;
    }

    toJSON(data?: any) {
        data = typeof data === 'object' ? data : {};
        data["rootId"] = this.rootId;
        data["companyId"] = this.companyId;
        data["areaId"] = this.areaId;
        data["slsmanNameId"] = this.slsmanNameId;
        data["groupNum"] = this.groupNum;
        data["organization"] = this.organization;
        data["shopTotal"] = this.shopTotal;
        data["scanQuantity"] = this.scanQuantity;
        data["scanFrequency"] = this.scanFrequency;
        data["priceTotal"] = this.priceTotal;
        data["custIntegral"] = this.custIntegral;
        data["retailerIntegral"] = this.retailerIntegral;
        data["creationTime"] = this.creationTime;
        data["specification"] = this.specification;
        return data;
    }

    clone() {
        const json = this.toJSON();
        let result = new ShopReportData();
        result.init(json);
        return result;
    }
}
export interface IShopReportData {
    rootId: number;
    companyId: number;
    areaId: number;
    slsmanNameId: number;
    groupNum: number;
    organization: string;
    shopTotal: number;
    scanQuantity: number;
    scanFrequency: number;
    priceTotal: number;
    custIntegral: number;
    retailerIntegral: number;
    creationTime: Date;
    specification: string;
}